lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -307,16 +307,25 @@ def __get_k8s_containers(self, ajs):
             envs = []
             if 'requests' in resources:
                 requests = resources['requests']
-                if 'memory' in requests:
-                    envs.append({'name': 'GALAXY_MEMORY_MB', 'value': str(ByteSize(requests['memory']).to_unit('M', as_string=False))})
                 if 'cpu' in requests:
-                    envs.append({'name': 'GALAXY_SLOTS', 'value': str(int(math.ceil(float(requests['cpu']))))})
+                    cpu_val = int(math.ceil(float(requests['cpu'])))
+                    envs.append({'name': 'GALAXY_SLOTS', 'value': str(cpu_val)})
+                if 'memory' in requests:
+                    mem_val = ByteSize(requests['memory']).to_unit('M', as_string=False)
+                    envs.append({'name': 'GALAXY_MEMORY_MB', 'value': str(mem_val)})
+                    if cpu_val:
```

Comment:
`cpu_val` is possibly unbound here if there is no `cpu in requests`. I think you can set `cpu_val = None` before line 310.

Reply:
Resolved in https://github.com/galaxyproject/galaxy/pull/11227/commits/c71db95d83e9673ca0dfa0a811be4611b0eb1c1d sorry for the oversight :)